numpy/core/src/umath/loops.c.src

```diff
@@ -95,6 +97,28 @@
  * combine with NPY_GCC_OPT_3 to allow autovectorization
  * should only be used where its worthwhile to avoid code bloat
  */
+#define BASE_OUTPUT_LOOP(tout, op) \
+    OUTPUT_LOOP { \
+        tout * out = (tout *)op1; \
+        op; \
+    }
+#define OUTPUT_LOOP_FAST(tout, op) \
+    do { \
+    /* condition allows compiler to optimize the generic macro */ \
+    if (IS_OUTPUT_CONT(tout)) { \
+        if (args[0] == args[1]) { \
```

Comment:
This inner `if` is meaningless - you're ignoring the input, so there's no point checking if its the same as the output.

It's there in the unary case to allow `a = -a` to be optimized differently from `b = -a`. There's no difference between `a = 0` and `b = 0` to the optimizer.

Reply:
Thanks, removed